tests/integration/Render/RenderTest.php

```diff
@@ -0,0 +1,146 @@
+				'__version' => 1,
+				'display_name' => 'Test ZIP API',
+				'endpoint' => 'https://wpvip.com/not-a-real-api',
+			],
+		] );
+
+		$test_query_runner = self::get_query_runner_with_response([
+			'post code' => 12345,
+			'places' => [
+				[
+					'place name' => 'Test City',
+					'state' => 'Test State',
+				],
+			],
+		]);
+
+		$test_query = HttpQuery::from_array( [
+			'data_source' => $test_data_source,
+			'query_runner' => $test_query_runner,
+			'input_schema' => [
+				'zip_code' => [
+					'name' => 'Zip Code',
+					'type' => 'string',
+				],
+			],
+			'output_schema' => [
+				'is_collection' => false,
+				'type' => [
+					'zip_code' => [
+						'name' => 'Zip Code',
+						'path' => '$["post code"]',
+						'type' => 'string',
+					],
+					'city' => [
+						'name' => 'City',
+						'path' => '$.places[0]["place name"]',
+						'type' => 'string',
+					],
+					'state' => [
+						'name' => 'State',
+						'path' => '$.places[0].state',
+						'type' => 'string',
+					],
+				],
+			],
+		] );
+
+		$registration_result = register_remote_data_block( [
+			'title' => 'Test ZIP API',
+			'render_query' => [
+				'query' => $test_query,
+			],
+		] );
+
+		$this->assertTrue( $registration_result );
+
+		// Register block configuration with WordPress, normally done during the 'init' filter
+		$block_config = ConfigStore::get_block_configuration( 'remote-data-blocks/test-zip-api' );
+		$this->assertTrue( is_array( $block_config ) && [] !== $block_config );
+		BlockRegistration::register_block_configuration( $block_config );
+
+		$result_html = do_blocks('
+			<!-- wp:remote-data-blocks/test-zip-api {"remoteData":{"blockName":"remote-data-blocks/test-zip-api","queryInput":{"zip_code":"12345"}}} -->
+			<div class="wp-block-remote-data-blocks-test-zip-api rdb-container">
+				<!-- wp:heading {"metadata":{"bindings":{"content":{"source":"remote-data/binding","args":{"block":"remote-data-blocks/test-zip-api","field":"zip_code"}}},"name":"Zip Code"}} -->
+				<h2 id="field-zip-code" class="wp-block-heading"></h2>
+				<!-- /wp:heading -->
+
+				<!-- wp:paragraph {"metadata":{"bindings":{"content":{"source":"remote-data/binding","args":{"block":"remote-data-blocks/test-zip-api","field":"city"}}},"name":"City"}} -->
+				<p id="field-city"></p>
+				<!-- /wp:paragraph -->
+
+				<!-- wp:paragraph {"metadata":{"bindings":{"content":{"source":"remote-data/binding","args":{"block":"remote-data-blocks/test-zip-api","field":"state"}}},"name":"State"}} -->
+				<p id="field-state"></p>
+				<!-- /wp:paragraph -->
+			</div>
+			<!-- /wp:remote-data-blocks/test-zip-api -->
+		');
+
+		$dom = self::load_html_into_dom( $result_html );
+
+		$this->assertIdInDomHasContent( $dom, 'field-zip-code', '12345' );
+		$this->assertIdInDomHasContent( $dom, 'field-city', 'Test City' );
+		$this->assertIdInDomHasContent( $dom, 'field-state', 'Test State' );
+	}
+
+	private function get_query_runner_with_response( array $response_data ) {
```

Comment:
This kindof works like [`MockQueryRunner`](https://github.com/Automattic/remote-data-blocks/blob/8242f1444412d6112372c55cdfe9ce76d3197a07/tests/inc/Mocks/MockQueryRunner.php), but I wanted to make it easier to provide a JSON-like data result with multiple data values like it's used above. We could extend `MockQueryRunner` to provide this interface instead of single-value results, but I'm not sure if it makes sense to have another mock interface for this level of code.